PostToExternalServices.py

```diff
@@ -0,0 +1,155 @@
+import json
+import logging
+import os
+
+import boto3
+import requests
+import tweepy
+
+AUTH0_BASE = "https://calligre.auth0.com"
+FB_BASE = "https://graph.facebook.com/v2.7"
+S3_BUCKET = "calligre-media"
+
+s3_client = boto3.client('s3')  # pylint: disable=C0103
+
+secrets = {}  # pylint: disable=C0103
+
+log = logging.getLogger()  # pylint: disable=C0103
+log.setLevel(logging.DEBUG)
+
+
+def set_secrets():
+    log.info("Fetching secrets from S3")
+    try:
+        response = s3_client.get_object(
+            Bucket="calligre-config",
+            Key="secrets.json",
+        )
+    except Exception as ex:
+        log.error(ex)
+        raise ex
+
+    content = response.get("Body").read()
+    retrieved_secrets = json.loads(content)
+    if len(retrieved_secrets) == 0:
+        log.error("Didn't fetch any secrets!")
+        raise Exception
+    elif len(retrieved_secrets) < 4:
+        log.warning("Number of secrets seems oddly small...")
+    secrets.update(retrieved_secrets)
+
+
+def set_auth0_token():
+    log.info("Fetching Auth0 token")
+    if not (secrets.get("AUTH0_CLIENT_ID") and
+            secrets.get("AUTH0_CLIENT_SECRET")):
+        log.error("Auth0 Client Tokens not set!")
+        raise Exception
+    payload = {
+        "client_id": secrets.get("AUTH0_CLIENT_ID"),
+        "client_secret": secrets.get("AUTH0_CLIENT_SECRET"),
+        "audience": "%s/api/v2/" % AUTH0_BASE,
+        "grant_type": "client_credentials"
+    }
+    try:
+        token = requests.post("%s/oauth/token" % AUTH0_BASE, json=payload).\
+            json()
+    except Exception as ex:
+        log.error(ex)
+        raise ex
+    secrets["AUTH0_TOKEN"] = token.get("access_token")
+    if not secrets["AUTH0_TOKEN"]:
+        log.error("Failed to fetch Auth0 Token!")
+        log.error(token)
+        raise Exception
```

Comment:
More specific in all exceptions we raise manually. Makes it so much easier to debug shit.